### ISSUE TITLE
V6 MTK6886 Error (Preloader option Error)

### ISSUE THREAD

**Issue by macekboris** [bkerler/mtkclient#1075]
Hi, the fix now works under Win11 environment.

Tried to test with printgp and use of engineriing preloader and DA loader.

Here is the resoult :

python mtk.py printgpt
MTK Flash/Exploit Client Public V2.0.1 (c) B.Kerler 2018-2024

Preloader - Status: Waiting for PreLoader VCOM, please reconnect mobile to brom mode

Power off the phone before connecting.
For brom mode, press and hold vol up, vol dwn, or all hw buttons and connect usb.
For preloader mode, don't press any hw button and connect usb.
If it is already connected and on, hold power for 10 seconds to reset.

Port - Device detected :)
Preloader -     CPU:                    MT6886(Dimensity 7200 Ultra)
Preloader -     HW version:             0x0
Preloader -     WDT:                    0x1c007000
Preloader -     Uart:                   0x11002000
Preloader -     Brom payload addr:      0x100a00
Preloader -     DA payload addr:        0x2001000
Preloader -     CQ_DMA addr:            0x10212000
Preloader -     Var1:                   0xa
Preloader - Disabling Watchdog...
Preloader - HW code:                    0x1229
Preloader - Target config:              0xe7
Preloader -     SBC enabled:            True
Preloader -     SLA enabled:            True
Preloader -     DAA enabled:            True
Preloader -     SWJTAG enabled:         True
Preloader -     EPP_PARAM at 0x600 after EMMC_BOOT/SDMMC_BOOT:  False
Preloader -     Root cert required:     False
Preloader -     Mem read auth:          True
Preloader -     Mem write auth:         True
Preloader -     Cmd 0xC8 blocked:       True
Preloader - Get Target info
Preloader - BROM mode detected.
Preloader -     HW subcode:             0x8a00
Preloader -     HW Ver:                 0xca01
Preloader -     SW Ver:                 0x1
Preloader - ME_ID:                      821A561FB03FB1269C898ED9F02836FC
DaHandler - Device is protected.
DaHandler - Device is in BROM-Mode. Bypassing security.
DaHandler
DaHandler - [LIB]: ←[31mFailed to dump preloader from ram, provide a valid one via --preloader option←[0m
DAXML - Uploading xflash stage 1 from MTK_DA_V6.bin
Preloader - SLA required ...

-----
After adding the path to eng. preloader and DA loader :

C:\mtkclient>python mtk.py printgpt --preloader=Loader/Preloader/preloader_zircon.bin
MTK Flash/Exploit Client Public V2.0.1 (c) B.Kerler 2018-2024

Traceback (most recent call last):
  File "C:\mtkclient\mtk.py", line 980, in <module>
    mtk = Main(args).run(parser)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 352, in run
    ArgHandler(self.args, config)
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 104, in __init__
    self.info("O:Var1:\t\t" + args.var1)
              ~~~~~~~~~~~~~~^~~~~~~~~~~
TypeError: can only concatenate str (not "NoneType") to str

-----
C:\mtkclient>python mtk.py printgpt --preloader=Loader/Preloader/preloader_zircon.bin --loader=Loader/DA_BR.bin
MTK Flash/Exploit Client Public V2.0.1 (c) B.Kerler 2018-2024

Traceback (most recent call last):
  File "C:\mtkclient\mtk.py", line 980, in <module>
    mtk = Main(args).run(parser)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 352, in run
    ArgHandler(self.args, config)
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 104, in __init__
    self.info("O:Var1:\t\t" + args.var1)
              ~~~~~~~~~~~~~~^~~~~~~~~~~
TypeError: can only concatenate str (not "NoneType") to str

ping @bkerler

**Comment by macekboris** [comment]
as saw allready exists as https://github.com/bkerler/mtkclient/issues/1072#issue-2367861104

**Comment by macekboris** [comment]
C:\mtkclient>python mtk.py r boot boot.bin --preloader=C:\mtkclient\mtkclient\Loader\Preloader\preloader_zircon.bin
MTK Flash/Exploit Client Public V2.0.1 (c) B.Kerler 2018-2024

Traceback (most recent call last):
  File "C:\mtkclient\mtk.py", line 980, in <module>
    mtk = Main(args).run(parser)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 352, in run
    ArgHandler(self.args, config)
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 104, in __init__
    self.info("O:Var1:\t\t" + args.var1)
              ~~~~~~~~~~~~~~^~~~~~~~~~~
TypeError: can only concatenate str (not "NoneType") to str

C:\mtkclient>

**Comment by bkerler** [comment]
Fixed in [36eec49](https://github.com/bkerler/mtkclient/commit/36eec4904523702a78d01c6c6f1dfe97e095ac0a)

**Comment by macekboris** [comment]
Used new commit :

C:\mtkclient>python mtk.py r boot boot.bin --preloader=C:\mtkclient\mtkclient\Loader\Preloader\preloader_zircon.bin
MTK Flash/Exploit Client Public V2.0.1 (c) B.Kerler 2018-2024

Traceback (most recent call last):
  File "C:\mtkclient\mtk.py", line 980, in <module>
    mtk = Main(args).run(parser)
          ^^^^^^^^^^^^^^^^^^^^^^
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 352, in run
    ArgHandler(self.args, config)
  File "C:\mtkclient\mtkclient\Library\mtk_main.py", line 104, in __init__
    self.info("O:Var1:\t\t" + config.chipconfig.var1)
              ~~~~~~~~~~~~~~^~~~~~~~~~~~~~~~~~~~~~~~
TypeError: can only concatenate str (not "int") to str

C:\mtkclient>
